From 0071a45a353dcca16d14cab373f84f6f3f086700 Mon Sep 17 00:00:00 2001
From: Eric Long <i@hack3r.moe>
Date: Fri, 17 May 2024 18:24:37 +0000
Subject: [PATCH] Fix Rust target triple for riscv64
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Rust target triple for riscv64 is riscv64gc-unknown-linux-gnu rather
than Clang's riscv64-unknown-linux-gnu.

Change-Id: I2f19101bd654f2f71392a65ce619c30570cdea49
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5237789
Commit-Queue: Łukasz Anforowicz <lukasza@chromium.org>
Reviewed-by: danakj <danakj@chromium.org>
Reviewed-by: Łukasz Anforowicz <lukasza@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1302721}
---
 AUTHORS               | 1 +
 build/config/rust.gni | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/AUTHORS b/AUTHORS
index d1911a8c224df..61c46e853b230 100644
--- a/src/AUTHORS
+++ b/src/AUTHORS
@@ -400,6 +400,7 @@ Emil Suleymanov <emil@esnx.xyz>
 Ergun Erdogmus <erdogmusergun@gmail.com>
 Eric Ahn <byungwook.ahn@gmail.com>
 Eric Huang <ele828@gmail.com>
+Eric Long <i@hack3r.moe>
 Eric Rescorla <ekr@rtfm.com>
 Erik Hill <erikghill@gmail.com>
 Erik Kurzinger <ekurzinger@gmail.com>
diff --git a/build/config/rust.gni b/build/config/rust.gni
index e99d7d08e1c1b..43ebdd5f73ac6 100644
--- a/src/build/config/rust.gni
+++ b/src/build/config/rust.gni
@@ -206,6 +206,8 @@ if (is_linux || is_chromeos) {
     } else {
       rust_abi_target = "arm-unknown-linux-gnueabi" + float_suffix
     }
+  } else if (current_cpu == "riscv64") {
+    rust_abi_target = "riscv64gc-unknown-linux-gnu"
   } else {
     # Best guess for other future platforms.
     rust_abi_target = current_cpu + "-unknown-linux-gnu"
